### PR TITLE
Added restart flag for services

### PR DESCRIPTION
**Pull request by jnummelin**
This PR adds restart flag for service containers.

Currently the defaulting to `always` policy happens in agent, would it make more sense to have it on master model level?

### CODE REVIEW

**Comment by jakolehm** [comment]
Yes, default should come from master.

One thing missing from PR is how master handles service instance failures if restart-policy is something else than `always`. Now if it sees that some instances are not properly running it will try to re-redeploy them.

**Comment by jnummelin** [comment]
Yes, absolutely. Kinda forgot to slab WIP flag on this. :)

**Comment by jnummelin** [comment]
@jakolehm Added the handling for balancing job. I think it should play pretty well with the docker restart policy mechanism: https://docs.docker.com/engine/reference/run/#restart-policies-restart

So only services with policies `always` or `unless-stopped` should be restarted/balanced by the master.

File: server/app/mutations/grid_services/create.rb [hunk 1]
```diff
@@ -113,6 +114,10 @@ def validate
       if self.strategy && !self.strategies[self.strategy]
         add_error(:strategy, :invalid_strategy, 'Strategy not supported')
       end
+      @re ||= Regexp.union(/^no$/, /^on-failure/, /^always$/, /^unless-stopped$/)
```
Comment (jakolehm) [inline server/app/mutations/grid_services/create.rb]: I think we don't need instance variable here?

Comment (jnummelin) [inline server/app/mutations/grid_services/create.rb]: True, I'll remove that.

**Comment by jakolehm** [comment]
I think we need to indicate service state somehow to user so that "fully completed" services do not show as "running".

**Comment by jnummelin** [comment]
I think service restarting needs anyways bigger thinking going forward. I'll close this one as it's getting dusty anyway...

**Pull request closed without merging**